crates/biome_js_analyze/src/lint/nursery/use_google_font_preconnect.rs

```diff
@@ -0,0 +1,141 @@
+    type Query = Ast<AnyJsxElement>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        if node.name().ok()?.name_value_token()?.text_trimmed() != "link" {
+            return None;
+        }
+
+        let href = node.get_attribute_inner_string_text("href")?;
+        let rel = node.get_attribute_inner_string_text("rel");
+
+        if href.starts_with("https://fonts.gstatic.com") && (rel.is_none() || rel? != "preconnect")
+        {
+            return Some(node.syntax().text_range());
+        }
+
+        None
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, range: &Self::State) -> Option<RuleDiagnostic> {
+        return Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                range,
+                markup! {
+                    "Attribute "<Emphasis>"rel=\"preconnect\""</Emphasis>" is missing from Google Font link."
+                },
+            )
+            .note(markup!{
+                "Not using "<Emphasis>"preconnect"</Emphasis>" can cause slower font loading and increase latency."
+            })
+        );
+    }
+
+    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+
+        if node.find_attribute_by_name("rel").is_some() {
+            return None;
+        }
+
+        let mut mutation = ctx.root().begin();
+        let mut attributes: Vec<_> = node.attributes().iter().collect();
+
+        let last_attr_token = match attributes.last()? {
+            AnyJsxAttribute::JsxAttribute(a) => a.name_value_token()?,
+            AnyJsxAttribute::JsxSpreadAttribute(a) => a.l_curly_token().ok()?,
+        };
+
+        let rel = if last_attr_token.has_leading_whitespace_or_newline() {
+            let pieces = last_attr_token.leading_trivia().pieces();
+            make::jsx_ident("rel").with_leading_trivia_pieces(pieces)
+        } else {
+            make::jsx_ident("rel").with_leading_trivia([(TriviaPieceKind::Whitespace, " ")])
+        };
+
+        let new_attribute = AnyJsxAttribute::from(
+            make::jsx_attribute(AnyJsxAttributeName::JsxName(make::jsx_name(rel)))
+                .with_initializer(make::jsx_attribute_initializer_clause(
+                    make::token(T![=]),
+                    AnyJsxAttributeValue::JsxString(make::jsx_string(make::jsx_string_literal(
+                        "preconnect",
+                    ))),
+                ))
+                .build(),
+        );
+
+        attributes.push(new_attribute);
+        mutation.replace_node(node.attributes(), make::jsx_attribute_list(attributes));
+
+        Some(JsRuleAction::new(
+            ActionCategory::QuickFix,
+            ctx.metadata().applicability(),
+            markup! { "Add "<Emphasis>"rel=\"preconnect\""</Emphasis>" attribute." }.to_owned(),
```

Comment:
```suggestion
            markup! { "Add the "<Emphasis>"rel=\"preconnect\""</Emphasis>" attribute." }.to_owned(),
```

You'll have to update the snapshots if you accept the change